gunrock/oprtr/intersection/kernel.cuh

```diff
@@ -164,38 +164,69 @@ __launch_bounds__(Dispatch<FLAG, InKeyT, OutKeyT, SizeT, ValueT, VertexT,
                                           num_vertex, num_edge, inter_op);
 }
 
+template <typename GraphT, bool VALID>
+struct GraphT_Switch {
+  template <OprtrFlag FLAG, typename FrontierInT, typename FrontierOutT,
+            typename ParametersT, typename InterOpT>
+  static cudaError_t Launch_CSR(const GraphT &graph,
+                                const FrontierInT *frontier_in,
+                                FrontierOutT *frontier_out,
+                                ParametersT &parameters, InterOpT inter_op) {
+    return util::GRError(cudaErrorInvalidDeviceFunction,
+                         "Intersect is not implemented for the given "
+                         "graph representation.");
+  }
+};
+
+template <typename GraphT>
+struct GraphT_Switch<GraphT, true> {
+  template <OprtrFlag FLAG, typename FrontierInT, typename FrontierOutT,
+            typename ParametersT, typename InterOpT>
+  static cudaError_t Launch_CSR(const GraphT &graph,
+                                const FrontierInT *frontier_in,
+                                FrontierOutT *frontier_out,
+                                ParametersT &parameters, InterOpT inter_op) {
+    typedef typename GraphT ::CsrT CsrT;
+    typedef typename FrontierInT ::ValueT InKeyT;
+    typedef typename FrontierOutT::ValueT OutKeyT;
+    typedef typename ParametersT ::SizeT SizeT;
+    typedef typename ParametersT ::ValueT ValueT;
+    typedef typename ParametersT ::VertexT VertexT;
+    typedef typename ParametersT ::LabelT LabelT;
+    typedef typename Dispatch<FLAG, InKeyT, OutKeyT, SizeT, ValueT, VertexT,
+                              InterOpT, true>::KernelPolicyT KernelPolicyT;
```

Comment:
Consider `using` instead of `typedef` for readability.

>using CsrT = typename GraphT::CsrT
using InKeyT = typename FrontierInT::ValueT
.... etc.
using KernelPolicyT = typename Dispatch<FLAG, InKeyT, O.......

Reply:
Discussed with @porumbes and we will leave this unchanged for now for consistency with other gunorck files.